File: backend/src/main/java/com/ballastlane/pokemon/domain/model/Pokemon.java
package com.ballastlane.pokemon.domain.model;

import java.util.List;

public record Pokemon(
        Long id,
        String name,
        Integer height,
        Integer weight,
        String category,
        String spriteUrl,
        List<String> skills
) {
    public Pokemon(
            Long id,
            String name,
            Integer height,
            Integer weight,
            String category,
            String spriteUrl,
            String localizedName,
            String locationMetadata,
            List<String> tags,
            List<String> skills
    ) {
        this(id, name, height, weight, category, spriteUrl, skills);
    }
}

File: frontend/src/types/pokemon.ts
export interface PokemonSummary {
  id: number;
  name: string;
  height: number;
  weight: number;
  category: string;
  spriteUrl: string;
  skills: string[];
}

export interface PokemonPage {
  items: PokemonSummary[];
  totalElements: number;
  pageNumber: number;
  pageSize: number;
}

export interface EvolutionNode {
  speciesName: string;
  minLevel: number | null;
  trigger: string | null;
  evolvesTo: EvolutionNode[];
}

export interface PokemonDetail extends PokemonSummary {
  description: string;
  stats: Record<string, number>;
  sprites: Record<string, string>;
  evolutionChain: EvolutionNode[];
}

export interface PokemonLocal {
  id: number;
  name: string;
  height: number;
  weight: number;
  category: string;
  spriteUrl: string;
  localizedName: string | null;
  locationMetadata: string | null;
  tags: string[];
}
